internal/app/server/server.go

```diff
@@ -106,6 +129,9 @@ func registerShutdownHandler(
 			logger.Info(ctx, "initiating grpc graceful stop")
 			gracefulStop()
 			_ = logger.Sync()
+			if shutdownErr := adminShutdown(ctx); shutdownErr != nil {
+				logger.With("err", shutdownErr).Error(ctx, "admin shutdown error: ", shutdownErr.Error())
+			} // TODO(lisalu): Where to place admin shutdown?
```

Comment:
I think how you have it is good. I'd suggest shutting it down before `gracefulStop()` is called though, otherwise orchestrator and other modules may have cleaned up things required that the admin server still requires.